s3store/s3store.go

```diff
@@ -148,7 +149,7 @@ func New(bucket string, service S3API) S3Store {
 	return S3Store{
 		Bucket:      bucket,
 		Service:     service,
-		MaxPartSize: 6 * 1024 * 1024,
+		MaxPartSize: 5 * 1024 * 1024 * 1024,
```

Comment:
I would also adapt the tests to test for this limit, so mistakes like that one can not happen anymore.